src/components/toast-notifications/toastNotifications.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+angular.module('openshiftCommonUI')
+  .directive('toastNotifications', function(NotificationsService, $timeout) {
+    return {
+      restrict: 'E',
+      templateUrl: 'src/components/toast-notifications/toast-notifications.html',
+      link: function($scope) {
+        $scope.notifications = NotificationsService.getNotifications();
+
+        $scope.close = function(notification) {
+          notification.hidden = true;
+          if (_.isFunction(notification.onClose)) {
+            notification.onClose();
+          }
+        };
+        $scope.onClick = function(notification, link) {
+          if (_.isFunction(link.onClick)) {
+            // If onClick() returns true, also hide the alert.
+            var close = link.onClick();
+            if (close) {
+              notification.hidden = true;
+            }
+          }
+        };
+        $scope.setHover = function(notification, isHover) {
+          notification.isHover = isHover;
+        };
+
+        $scope.$watch('notifications', function() {
+          _.forEach($scope.notifications, function(notification) {
```

Comment:
nit: a while back we decided to use _.each instead of _.forEach and everywhere for consistency